File: module/AudioPlayBack.py
__author__ = 'toramisu'
from PyQt5.QtMultimedia import QMediaPlayer, QMediaContent
from PyQt5.QtCore import QUrl
from module.Events import *


class AudioPlayBack():
    def __init__(self):
        self.__audioPlayback = QMediaPlayer()
        self.__audioPlayback.setVolume(50)
        self.__audioPlayback.positionChanged.connect(self.onTick)

        pass

    def onTick(self, e=None):
        print(self.__audioPlayback.position(), self.__audioPlayback.state())
        Event.dis(AudioPlaybackEvent.TICK, self.__audioPlayback.position())
        pass

    def load(self, path):
        content = QMediaContent(QUrl(path))
        self.__audioPlayback.setMedia(content)
        self.__audioPlayback.play()
        pass

    def seek(self, position):
        self.__audioPlayback.setPosition(position)
        pass

    def play(self):
        if self.__audioPlayback.state() == QMediaPlayer.PausedState:
            self.__audioPlayback.play()
            pass
        pass

    def pause(self):
        if self.__audioPlayback.state() == QMediaPlayer.PlayingState:
            self.__audioPlayback.pause()
            pass
        pass

    # 0~100
    def setVolume(self, vol):
        self.__audioPlayback.setVolume(vol)
        pass

File: model/playback/AudioPlayback.cpp
//
// Created by manoyuria on 2015/6/19.
//

#include "AudioPlayback.h"
#include "events/Event.h"

AudioPlayback::AudioPlayback() {
    audioplayback = new QMediaPlayer();
    audioplayback->setVolume(50);
    audioplayback->connect(audioplayback, QMediaPlayer::positionChanged, [this]{this->onPosition();});
}

void AudioPlayback::onPosition() {
//    qDebug() << audioplayback->position() << audioplayback->state();
    Evt_dis(AUDIO_TICK);
}

void AudioPlayback::load(QString path) {
    audioplayback->setMedia(QMediaContent(QUrl(path)));
}

void AudioPlayback::seek(int position) {
    audioplayback->setPosition(position);
}

void AudioPlayback::play() {
    if (audioplayback->state() != QMediaPlayer::PlayingState) {
        audioplayback->play();
    }
}

void AudioPlayback::pause() {
    if (audioplayback->state() == QMediaPlayer::PlayingState) {
        audioplayback->pause();
    }
}

void AudioPlayback::setVolume(int vol) {
    audioplayback->setVolume(vol);
}

void AudioPlayback::stop() {

}
